kits/arm/example_AR_kit.cpp

```diff
@@ -0,0 +1,200 @@
+  Eigen::VectorXd home_position(arm -> robotModel().getDoFCount());
+  home_position << 0, M_PI/3, 2*M_PI/3, 5*M_PI/6, -M_PI/2, 0;
+
+  // Command the softstart to home position
+  arm -> update();
+  arm -> setGoal(arm::Goal::createFromPosition(4, home_position)); // take 4 seconds
+  arm -> send();
+
+  // Get the cartesian position and rotation matrix @ home position
+  Eigen::Vector3d xyz_home; 
+  Eigen::Matrix3d rot_home;
+  arm -> FK(home_position, xyz_home, rot_home); 
+
+  // Set up states for the mobile device
+  Eigen::Vector3d xyz_phone_init;
+  Eigen::Matrix3d rot_phone_init;
+  Eigen::Matrix3d rot_init_target;//(3,3);
+  rot_init_target << 0,-1,0, 1,0,0, 0,0,1;
+
+  // Target variables
+  Eigen::VectorXd target_joints(arm -> robotModel().getDoFCount());
+
+  while(arm->update())
+  {
+
+    if (softstart) {
+      Eigen::VectorXd home_position(arm -> robotModel().getDoFCount());
+      // End softstart when arm reaches its homePosition
+      if (arm -> atGoal()){
+        mobile -> sendText("Softstart Complete!");
+        softstart = false; 
+        continue;
+      }
+      arm -> send();
+
+      // Stay in softstart, don't do any other behavior
+      continue;
+    } 
+
+    // Get latest mobile_state
+    auto mobile_state = mobile->getState();
+    MobileIODiff diff(last_mobile_state, mobile_state);
+
+    // Get latest gripper_state
+    // gripper -> getState();
+
+    // Button B1 - Return to home position
+    if (diff.get(1) == MobileIODiff::ButtonState::ToOn) {
+        ar_mode = false;
+        arm -> setGoal(arm::Goal::createFromPosition(4, home_position));
+    }
+
+    // Button B3 - Start AR Control
+    if (diff.get(3) == MobileIODiff::ButtonState::ToOn) {
+      xyz_phone_init << mobile -> getLastFeedback().mobile().arPosition().get().getX(),
+                        mobile -> getLastFeedback().mobile().arPosition().get().getY(),
+                        mobile -> getLastFeedback().mobile().arPosition().get().getZ();
+      std::cout << xyz_phone_init << std::endl;
+      rot_phone_init = makeRotationMatrix(mobile -> getLastFeedback().mobile().arOrientation().get());
+      ar_mode = true;
+    }
+
+    // Button B6 - Grav Comp Mode
+    if (diff.get(6) == experimental::MobileIODiff::ButtonState::ToOn) {
+      arm -> cancelGoal();
+      ar_mode = false;
+    }
+
+    // Button B8 - End Demo
+    if (diff.get(8) == experimental::MobileIODiff::ButtonState::ToOn) {
+      // Clear MobileIO text
+      mobile -> clearText();
+      return 1;
+    }
+
+    if (ar_mode) {
+      // Get the latest mobile position and orientation
+      Eigen::Vector3d phone_pos;
+      phone_pos << mobile -> getLastFeedback().mobile().arPosition().get().getX(),
+                   mobile -> getLastFeedback().mobile().arPosition().get().getY(),
+                   mobile -> getLastFeedback().mobile().arPosition().get().getZ();
+      auto phone_ori = mobile -> getLastFeedback().mobile().arOrientation().get();
```

Comment:
General comment -- I'm not used to this style (spaces between `mobile` and `->` and `getLastFeedback`.  I'd say go for a more condensed style, but I'd like @rLinks234 's opinion here.